File: src/main/kotlin/dev/jefy/connectpro/portfolio/applicaion/command/JobPostCommand.java
package dev.jefy.connectpro.portfolio.applicaion.command;


import org.jspecify.annotations.NullMarked;

import dev.jefy.connectpro.portfolio.applicaion.dtos.JobPostRequest;
import dev.jefy.connectpro.portfolio.domain.vo.JobPostId;

/**
 * @author Jôph Yamba
 */
@NullMarked
public interface JobPostCommand {
    
    JobPostId create(JobPostRequest request);
    JobPostId update(JobPostId jobPostId, JobPostRequest request);
    JobPostId open(JobPostId jobPostId);
    JobPostId close(JobPostId jobPostId);
}

File: src/main/kotlin/dev/jefy/connectpro/portfolio/application/command/JobPostCommand.kt
package dev.jefy.connectpro.portfolio.application.command

import dev.jefy.connectpro.portfolio.application.dtos.JobPostRequest
import dev.jefy.connectpro.portfolio.domain.vo.JobPostId
import org.jspecify.annotations.NullMarked

/**
 * @author Jôph Yamba
 */
@NullMarked
interface JobPostCommand {
    fun create(request: JobPostRequest): JobPostId
    fun update(jobPostId: JobPostId, request: JobPostRequest): JobPostId
    fun open(jobPostId: JobPostId): JobPostId
    fun close(jobPostId: JobPostId): JobPostId
    fun delete(jobPostId: JobPostId)
}
